### PR TITLE
docs: Separate Text MDX into separate MDXes

### PR DESCRIPTION
**Pull request by jamesfan**
## Summary

Breaks the Text MDX into separate MDXes for each Text component. This will allow us to consume the docs properly on the Canvas Site.

## Release Category
Documentation

---

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [standard MDX template](https://github.com/Workday/canvas-kit/discussions/1131)

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Where Should the Reviewer Start?

Review the Text component MDXes listed under `Components > Text` in the Storybook navigation.

I could have gone with a flatter hierarchy in Storybook, i.e.:

```
Components
  Text
    Body Text
    Text
```

... instead of what I ended up doing ...

```
Components
  Text
    Body Text
      React
        Basic
    Text
      React
        Basic
        Type Level
        Variant
```

... but I decided on the second approach since it allows us to create groups of examples for each Text component (and it's more consistent with how we document other components, Button being the exception).

### CODE REVIEW

File: modules/react/text/stories/BodyText.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+## Usage
+
+### Basic Example
+
+`BodyText` is a Type Level component (along with [`Title`](/components/text/title/),
+[`Heading`](/components/text/heading/), and [`Subtext`](/components/text/subtext/)) built on top of
+[`Text`](/components/text/text/) which provides a simple way to render body-level text according to
+the [Canvas type hierarchy](/tokens/type/#type-styles).
```
Comment (jamesfan) [inline modules/react/text/stories/BodyText.stories.mdx:27]: This link will not work properly in Storybook (the `#type-styles` doesn't correspond to an `id` on the destination page in SB), but it should work properly on the Canvas Site. There are a handful of other links in this PR which will have the same issue in SB.

File: modules/react/text/stories/LabelText.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+label.
+
+<ExampleCodeBlock code={Cursor} />
+
+### Disabled
+
+Set the `disabled` prop to provide a visual cue that the label and its associated element are
+disabled.
+
+<ExampleCodeBlock code={Disabled} />
```
Comment (jamesfan) [inline modules/react/text/stories/LabelText.stories.mdx:41]: I extracted Cursor and Disabled into separate examples.

File: modules/react/text/stories/Text.stories.mdx [hunk 3]
```diff
@@ -1,132 +1,54 @@
...
+understanding of the text being rendered.
 
-```tsx
-// v7
-<p css={{...type.levels.subtext.small}}>Small Subtext Text</p>
-
-// v8
-<Subtext size="small">Small Subtext Text</Subtext>
-```
+<ExampleCodeBlock code={Variant} />
```
Comment (jamesfan) [inline modules/react/text/stories/Text.stories.mdx:47]: I created additional examples for `typeLevel` and `variant`.

**Comment by jamesfan** [comment]
I would be nice if `Text` showed up at the top of the list since it's the foundational component for the rest of the Text components. Should be a way to do this via the [`storySort`](https://github.com/Workday/canvas-kit/blob/master/.storybook/preview.js#L19) function in `preview.js` but I opted not to spend too much time on this since it feels lower priority. It's easy to control the sorting order on the Canvas Site (though I believe the plan at the moment is to show them in alphabetical order over there).

File: modules/react/text/stories/Text.stories.mdx (hunk 3 above)
Comment (alanbsmith) [reply modules/react/text/stories/Text.stories.mdx:47]: Thank you!

**Review by alanbsmith — APPROVED** [review]
Great work! Thanks, James!

**Pull request merged**